Add manpage for qucspowercombining

andresmmera opened the pull request:
As mentioned in #860, the recently added qucspowercombining tool was
missing the man page file. This caused the cmake installation to crash.

Additionally, the content of the "Availabilty" and "Reporting bugs"
sections was put in the following file: /qucs/man/qucs_availability_bugs.1

felix-salfelder commented:
hmm. maybe only the variables? note that
PACKAGE_BUGREPORT and PACKAGE_URL already exist (these are standard autotools variables).

also make sure to use the configure.ac scripts corresponding to the package. the toplevel configure script is just band-aid (and maybe convenience).

andresmmera commented:
> hmm. maybe only the variables? note that
PACKAGE_BUGREPORT and PACKAGE_URL already exist (these are standard autotools variables).
 
uh, I don't see your point. Could you please ellaborate a bit more? Finally, I'm using three variables: QUCS_DESCRIPTION, QUCS_AVAILABILITY and QUCS_REPORT_BUGS.

> also make sure to use the configure.ac scripts corresponding to the package. the toplevel configure script is just band-aid (and maybe convenience).

Ok, but that implies duplicating the variables mentioned above. If I understood correctly, you're suggesting that we should define them both in qucs-core/configure.ac and  qucs/configure.ac and leave the top-level configure.ac untouched

On the other hand, the fact that the man pages are autogenerated using ./configure means that this approach won't work for the cmake build, right? I'll investigate if it exists a similar mechanism in cmake...

felix-salfelder commented:
On Sun, Sep 09, 2018 at 09:29:38AM -0700, Andrés Martínez-Mera wrote:
> > hmm. maybe only the variables? note that
> PACKAGE_BUGREPORT and PACKAGE_URL already exist (these are standard autotools variables).
>  
> uh, I don't see your point. Could you please ellaborate a bit more?
> Finally, I'm using three variables: QUCS_DESCRIPTION,
> QUCS_AVAILABILITY and QUCS_REPORT_BUGS.

i suggest to put something like

The latest version of Qucs can always be obtained from
\fB@PACKAGE_URL@\fR.

into the man page. yes it still duplicates some fragments, but not the
ones that require maintenance.

> > also make sure to use the configure.ac scripts corresponding to the
> > package. the toplevel configure script is just band-aid (and maybe
> > convenience).
> 
> Ok, but that implies duplicating the variables mentioned above. If I
> understood correctly, you're suggesting that we should define them
> both in qucs-core/configure.ac and  qucs/configure.ac and leave the
> top-level configure.ac untouched

yes. that's the price of having individual packages for individual
tasks. but the PACKAGE_* variables are already defined in every
*/configure.ac, so the overhead is quite small here.

> On the other hand, the fact that the man pages are autogenerated using
> ./configure means that this approach won't work for the cmake build,
> right?

yes. cmake is a can of worms on its own. the deal with merging
incomplete stuff: maintain it, or drop it, but i am biased. see #780.

thanks

andresmmera commented:
I proceeded as you suggested. 
Now I'll see what can we do with the CMake build.

felix-salfelder commented:
thanks

usually, AC_INIT is used to set these variables. *_BUGREPORT is already set (note that this should be an email address). maybe one URL is sufficient, i'd prefer https://sf.net/p/qucs over just sourceforge.net, YMMV

see https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Initializing-configure.html

andresmmera commented:
Please, excuse my ignorance about the autoconf stuff:

I put:
`AC_INIT([qucs],  [PACKAGE_URL], [https://sf.net/p/qucs])`
in configure.ac. However, when running ./bootstrap and ./configure, the latter gets stuck.

I'm trying to see how PACKAGE_NAME, PACKAGE_TARNAME, PACKAGE_BUGREPORT, etc. are set in the generated configure file..., but no success so far.

Specifically, I see that this line:
`AC_INIT([qucs],  [QUCS_VERSION], [qucs-bugs@lists.sourceforge.net])`
(configure.ac, line 9) sets PACKAGE_BUGREPORT='qucs-bugs@lists.sourceforge.net' in configure. But, honestly, I don't know how... Am I missing something trivial?

andresmmera commented:
@felix-salfelder Thanks a lot!!

felix-salfelder commented:
looks really good now. but when you rename the qucsator package in `qucs-core/configure.ac`, please rename it to "qucsator", that seems to make most sense to me.

i do not know the intent of calling it "qucs-core" in the first place. i can't see a core there, and i believe it happened in error, YMMV.

guitorri commented:
I will merge #863. When ready, please change the base branch of this PR and rebase on `release-0.0.20`.

guitorri commented:
Please revert 2110066 to get the manpages installed also with cmake.

andresmmera commented:
I have got 6 commits from felix and in3otd after the rebase against `release-0.0.20`, is this ok?

![image](https://user-images.githubusercontent.com/13180689/45583574-1e3a3780-b8c5-11e8-9ffc-05f864026f9e.png)

andresmmera commented:
Thanks, Felix, I'd forgotten such valuable command!!

Now, the commit history looks clean

![image](https://user-images.githubusercontent.com/13180689/45583881-25b00f80-b8ca-11e8-83d6-6b18f8788c9a.png)

in3otd commented:
not directly related but I see we keep forgetting to update the copyright notes - I'll open a separate issue for this.
Not that I consider this the utmost important thing but stating that [qucsactivefilter](https://github.com/Qucs/qucs/blob/develop/qucs/qucs-activefilter/qucsactivefilter.1#L36) was done 13 years ago by someone who never worked on it it simply plain wrong.

andresmmera commented:
I've pushed a fix for cmake...

guitorri commented:
Hum...
```
CMake Error at qucs/man/cmake_install.cmake:36 (file):
  file INSTALL cannot find "/home/travis/build/Qucs/qucs/qucs/man/qucs.1".
```

andresmmera commented:
Yes, I saw that... I can't take a look right now, maybe tomorrow or so...

andresmmera commented:
The cmake build passed :tada: :tada:

On the other hand, the clang build is failing. (I would swear it was green before...) 
After reviewing the build log, I don't see why it fails.

https://api.travis-ci.org/v3/job/430674247/log.txt

At the end, it says:

> The job exceeded the maximum log length, and has been terminated.

So I guess it's failing after that.

in3otd commented:
yep, I [also](https://github.com/Qucs/qucs/pull/865#issuecomment-422269246) saw this, don't know what changed in Travis.

guitorri commented:
Perhaps Travis bumped the clang version?
Try to disable the warning with `-Wno-error=unused-command-line-argument`
https://stackoverflow.com/questions/47094843/clang-warning-argument-unused-during-compilation-stdlib-libc

in3otd commented:
um, seems not
- this PR [clang version](https://travis-ci.org/Qucs/qucs/jobs/430674247#L1054)
- previous passing build [clang version](https://travis-ci.org/Qucs/qucs/jobs/428222754#L1044)

still `clang version 5.0.0 (tags/RELEASE_500/final)`

Maybe they reduced the max log length; those warnings are also in the previous passing build.

andresmmera commented:
I've made a diff of both logs. They are very similar indeed. The bad one stops at:

> mv -f .deps/comp_1bit.Tpo .deps/comp_1bit.Plo

whereas the good one continues to work...

![bad_vs_good](https://user-images.githubusercontent.com/13180689/45894415-8eb5ec80-bdce-11e8-8638-745add2bddff.jpg)

andresmmera commented:
I noticed that the bad one is a bit longer because, for example:

When it says:

> Cloning into 'Qucs/qucs'

 it shows the status of the download, which adds 398 extra lines.
![extra lines](https://user-images.githubusercontent.com/13180689/45895340-37654b80-bdd1-11e8-913a-7e07d578571e.png)


The same happens when it says:

> git fetch origin +refs/pull/862/merge:

which ads 326 extra lines...
![extra lines2](https://user-images.githubusercontent.com/13180689/45895383-582da100-bdd1-11e8-8fc7-6a23701c1a45.png)

My point is that the good one finished the qucs-core build by line 49501 (and starts with qucs-doc) whereas the bad one didn't finish that by the time it is forced to terminate (line 52595)

Can't we extend the maximum log length?

in3otd commented:
ops, modified the wrong files... 40ea1ee84939733804d2110305c1970d147f005a will need to be removed later

in3otd commented:
works now, there are still a bunch of warnings when compiling `adms` and `qucs-core`, some of which look pretty scary, like `warning: array index 2 is past the end of the array`, some coming also from autogenerated code but I have no idea if they are meaningful or not.

I've also noticed that the [test in configure.ac](https://github.com/Qucs/qucs/blob/d52358ca022ee2ae34b0f6886715b9b50b4eda54/qucs/configure.ac#L216) for the Qucs GUI does not work, as the same `CFLAGS` are applied for both [`gcc`](https://travis-ci.org/Qucs/qucs/jobs/431693826#L2753) and [`clang`](https://travis-ci.org/Qucs/qucs/jobs/431693825#L2086).
Maybe all this here should be put into a separate PR.

guitorri commented:
Please remove the last few Travis commits and rebase once more.
Hope it will be fine.

andresmmera commented:
Ok. Removed Travis-related commits and rebased.

![image](https://user-images.githubusercontent.com/13180689/46109790-ee9eff80-c1e1-11e8-9a7d-b0718511e0d9.png)
